src/raster/rasterband.rs

```diff
@@ -12,6 +15,68 @@ use ndarray::Array2;
 
 use crate::errors::*;
 
+/// Extra options used to read a raster.
+///
+/// For documentation, see `gdal_sys::GDALRasterIOExtraArg`.
+#[derive(Debug)]
+pub struct RasterIOExtraArg {
+    pub n_version: libc::c_int,
+    pub e_resample_alg: GDALRIOResampleAlg::Type,
+    pub pfn_progress: gdal_sys::GDALProgressFunc,
+    pub p_progress_data: *mut libc::c_void,
```

Comment:
We should hide the pointer stuff for now. In the long run we could add a callback and/or a future/channel to access the state.